File: content/examples/other/leetcode/problems/validate-binary-search-tree/example-1/solution.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def is_valid_bst_inorder_list(root):
    values = []

    def inorder(node):
        if not node:
            return
        inorder(node.left)
        values.append(node.val)
        inorder(node.right)

    inorder(root)
    for i in range(1, len(values)):
        if values[i] <= values[i - 1]:
            return False
    return True


if __name__ == "__main__":
    ok = TreeNode(2, TreeNode(1), TreeNode(3))
    print(is_valid_bst_inorder_list(ok))

File: content/examples/other/leetcode/problems/validate-binary-search-tree/example-1/solution.js
class TreeNode {
  constructor(val, left = null, right = null) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
}

function isValidBSTInorderList(root) {
  const values = [];
  (function inorder(node) {
    if (!node) return;
    inorder(node.left);
    values.push(node.val);
    inorder(node.right);
  })(root);

  for (let i = 1; i < values.length; i += 1) {
    if (values[i] <= values[i - 1]) return false;
  }
  return true;
}

if (require.main === module) {
  const ok = new TreeNode(2, new TreeNode(1), new TreeNode(3));
  console.log(isValidBSTInorderList(ok));
}

module.exports = { TreeNode, isValidBSTInorderList };
